src/test/java/io/neonbee/internal/verticle/HealthCheckVerticleTest.java

```diff
@@ -0,0 +1,55 @@
+package io.neonbee.internal.verticle;
+
+import static com.google.common.truth.Truth.assertThat;
+import static io.neonbee.internal.verticle.HealthCheckVerticle.SHARED_MAP_KEY;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.neonbee.data.DataQuery;
+import io.neonbee.data.DataRequest;
+import io.neonbee.internal.SharedDataAccessor;
+import io.neonbee.test.base.DataVerticleTestBase;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxTestContext;
+
+class HealthCheckVerticleTest extends DataVerticleTestBase {
+    private HealthCheckVerticle healthCheckVerticle;
+
+    @BeforeEach
+    @Timeout(value = 2, timeUnit = TimeUnit.SECONDS)
+    void setUp(VertxTestContext testContext) {
+        healthCheckVerticle = new HealthCheckVerticle();
+        deployVerticle(healthCheckVerticle).onComplete(testContext.succeedingThenComplete());
+    }
+
+    @Test
+    @Timeout(value = 2, timeUnit = TimeUnit.SECONDS)
+    void testRetrieveData(Vertx vertx, VertxTestContext testContext) {
+        DataRequest req = new DataRequest(healthCheckVerticle.getName(), new DataQuery());
+        Future<JsonArray> response = requestData(req);
+
+        assertData(response, resp -> {
+            assertThat(resp.size()).isEqualTo(1);
+
+            JsonObject firstCheck = resp.getJsonObject(0);
+            assertThat(firstCheck.containsKey("id")).isTrue();
+            assertThat(firstCheck.containsKey("status")).isTrue();
+            assertThat(firstCheck.containsKey("outcome")).isFalse();
+
+            new SharedDataAccessor(vertx, HealthCheckVerticleTest.class).getLock(SHARED_MAP_KEY)
+                    .onFailure(testContext::failNow).compose(lock -> getNeonBee().getAsyncMap().get(SHARED_MAP_KEY)
+                            .onComplete(testContext.succeeding(qualifiedNamesOrNull -> testContext.verify(() -> {
+                                String expectedName = healthCheckVerticle.getQualifiedName();
+                                assertThat((JsonArray) qualifiedNamesOrNull).containsExactly(expectedName);
+                            }))).onComplete(anyResult -> lock.release()));
+
+        }, testContext).onComplete(testContext.succeedingThenComplete());
+    }
```

Comment:
ad 1): yes, I can split it into 2 tests
ad 2): because HealthCheckVerticle is running twice on same node, but check results of same node have the same id and therefore are not added twice